src/services/selectRoofInteraction.ts

```diff
@@ -69,12 +72,24 @@ class SelectRoofInteraction extends AbstractInteraction {
     router.push({ name: 'roof-selected-information' })
   }
 
+  _isClickOnHomePageValid() {
+    const viewStore = useViewsStore()
+    if (viewStore.currentView !== viewList.home) {
+      return true
+    }
+    const homeRouter = useHomeStore()
+    if (homeRouter.isTermOfUseAccepted) {
+      return true
+    }
+    homeRouter.setDisplayError(true)
+    return false
+  }
+
   async pipe(event: InteractionEvent) {
     const selectedBuilding = event.feature
     const selectedBuildingId =
       selectedBuilding?.getProperty('attributes')['BUILDINGID']
-
-    if (selectedBuilding) {
```

Comment:
I agree with @MaYohann

Reply:
After a discussion with Yohann on slack, here is his answer: "You're right, it couldn't be easier, leave it like that
You would have to subscribe to both otherwise, and in one check that the other is ok, and in the other check that one is ok.
As a user will never 'uncheck' his box.
You don't need her to delete herself anyway, so it's not bad like that"